process/accounts/accounts_processor_test.go

```diff
@@ -269,8 +174,21 @@ func generateAddresses(n int) [][]byte {
 	return addresses
 }
 
-// This function only works if accounts.NewAccountsProcessor is called with
-// &mock.AccountsAdapterStub{UserAccountHandler: &mock.UserAccountMock{}}
+func prepareAlteredAccounts(addresses [][]byte) map[string]*indexer.AlteredAccount {
+	mapToRet := make(map[string]*indexer.AlteredAccount)
+
+	for idx, addr := range addresses {
+		mapToRet[string(addr)] = &indexer.AlteredAccount{
+			Balance: big.NewInt(0).SetInt64(int64(idx + 1)).String(),
+			Nonce:   big.NewInt(0).SetInt64(int64(idx + 1)).Uint64(),
+		}
+	}
+
+	return mapToRet
+}
+
+// This function relies on the order of the addresses according to generateAddresses(). for example, adr1 need to have
+// a balance of 2 and a nonce of 2
```

Comment:
I think we should change this addr**1** needs to have balance = **2** and nonce = **2**, so that we can have:
- addr**1** with  balance = **1** and nonce = **1** (otherwise it looks a bit confusing)

We can do this pretty easy by changing:
1.  `prepareAlteredAccounts` when setting balance and nonce, use `SetInt64(int64(idx))` instead of `SetInt64(int64(idx + 1))`
2. `checkProcessedAccounts` when checking  balance and nonce, instead of `addrIdx+1`, use `addrIdx`
3. Refactor `TestAccountsProcessor_ProcessAccounts_OneAddress_OneMetaChainShardAddress_ExpectOneAccount` to be consistent with other tests:

```go
	ap, _ := accounts.NewAccountsProcessor(&mock.ShardCoordinatorMock{})

	address := generateAddresses(1)
	tx := &schema.Transaction{
		Receiver: address[0],
		Sender:   utility.MetaChainShardAddress(),
	}

	alteredAccounts := prepareAlteredAccounts([][]byte{address[0]})
	ret := ap.ProcessAccounts(alteredAccounts, []*schema.Transaction{tx}, []*schema.SCResult{}, []*schema.Receipt{})

	require.Len(t, ret, 1)
	checkProcessedAccounts(t, address, ret)
```

Reply:
done